File: HNLineAnimationTextField-OC/HNLineAnimationTextFieldDemo-OC/HNLineAnimationTextField/Categories/UIWindow+HNHierarchy.m
//
//  UIWindow+HNHierarchy.m
//  HNLineAnimationTextFieldDemo-OC
//
//  Created by zakariyyaSv on 16/4/24.
//  Copyright © 2016年 Zakariyya. All rights reserved.
//

#import "UIWindow+HNHierarchy.h"

@implementation UIWindow (HNHierarchy)

- (UIViewController *)topMostController{
    UIViewController *topController = [self rootViewController];
    
    while ([topController presentedViewController]) {
        topController = [topController presentedViewController];
    }
    return topController;
}

- (UIViewController *)currentViewController{
    UIViewController *currentController = [self topMostController];
    while ([currentController isKindOfClass:[UINavigationController class]] && [(UINavigationController *)currentController topViewController]) {
        currentController = [(UINavigationController*)currentController topViewController];
    }
    return currentController;
}

@end

File: HNLineAnimationTextFieldDemo-swift/HNLineAnimationTextFieldDemo-swift/HNLineAnimationTextField/UIWindow+HNHierarchy.swift
//
//  UIWindow+HNHierarchy.swift
//  HNLineAnimationTextFieldDemo-swift
//
//  Created by zakariyyaSv on 16/4/26.
//  Copyright © 2016年 Zakariyya. All rights reserved.
//

import Foundation
import UIKit

public extension UIWindow{
     override public func topMostController() -> UIViewController?{
        var topController = rootViewController
        
        while let presentedController = topController?.presentedViewController {
            topController = presentedController;
        }
        return topController
        
    }
    
    public func currentViewController() -> UIViewController? {
        var  currentViewController = topMostController()
        while currentViewController != nil && currentViewController is UINavigationController && (currentViewController as! UINavigationController).topViewController != nil {
            currentViewController = (currentViewController as! UINavigationController).topViewController
        }
        return currentViewController
    }
}